forest/src/daemon.rs

```diff
@@ -204,12 +209,49 @@ pub(super) async fn start(config: Config) {
         .unwrap(),
     );
 
+    // Mining may or may not happen, depending on the consensus type.
+    let mining_task: Option<JoinHandle<anyhow::Result<()>>>;
+
+    // For consensus types that do mining, create a component to submit their proposals.
+    let submitter = SyncGossipSubmitter::new(
+        network_name.clone(),
+        network_send.clone(),
+        tipset_sink.clone(),
+    );
+
     // Initialize Consensus
-    type FullConsensus = FilecoinConsensus<DrandBeacon, FullVerifier>;
-    let consensus: FullConsensus = FilecoinConsensus::new(state_manager.beacon_schedule());
+    #[cfg(all(feature = "fil_cns", not(any(feature = "deleg_cns"))))]
```

Comment:
If neither `fil_cns` or `deleg_cns` is defined, could we print an error message?

Reply:
Good idea!